nucleus/metrics/filtering.py

```diff
@@ -0,0 +1,371 @@
+import enum
+import functools
+import logging
+from enum import Enum
+from typing import Callable, Iterable, List, NamedTuple, Sequence, Set, Union
+
+from nucleus.annotation import (
+    BoxAnnotation,
+    CategoryAnnotation,
+    CuboidAnnotation,
+    LineAnnotation,
+    MultiCategoryAnnotation,
+    PolygonAnnotation,
+    SegmentationAnnotation,
+)
+from nucleus.prediction import (
+    BoxPrediction,
+    CategoryPrediction,
+    CuboidPrediction,
+    LinePrediction,
+    PolygonPrediction,
+    SegmentationPrediction,
+)
+
+
+class FilterOp(str, Enum):
+    GT = ">"
+    GTE = ">="
+    LT = "<"
+    LTE = "<="
+    EQ = "="
+    EQEQ = "=="
```

Comment:
can we just put `==` here? They are both mapped to `==` anyways. I think having both of them is confusing.

Reply:
One is Python style and the other is SQL style, I think it is fine practice to support both, PyArrow does the same. It has no effect on maintainability but might save somebody a headache so I'll keep it in. 🙂